packages/langium-cli/src/generate.ts

```diff
@@ -27,6 +27,7 @@ const services = createLangiumGrammarServices();
 export async function generate(config: LangiumConfig): Promise<GeneratorResult> {
     const relPath = config[RelativePath];
     const absGrammarPath = URI.file(path.resolve(relPath, config.grammar));
+    services.documents.LangiumDocuments.invalidateDocument(absGrammarPath);
```

Comment:
It would be nice if we could reuse the same logic from the existing services.
Would `DocumentBuilder.documentChanged(uri: string)`  work here?
If not that we probably we need a better common API that does an update and returns a "fresh" Document.

Reply:
Well the CLI in a watch mode is a special case, that we definitely will resolve inside of Langium. However, for now (0.2.0), I just want to fix the `langium-cli`.

For the future, we want to provide utilities for watching files in a CLI. This will then also include correctly calling `documentChanged` in a more generic way.

Reply:
Could you describe that in a new issue?

Reply:
@spoenemann Done #239